lib/hammer_cli_import/configfile.rb

```diff
@@ -136,39 +136,52 @@ def clean_module(name)
       # inside of working-directory
       def generate_module_template_for(name)
         module_name = name
+        pwd = Dir.pwd
         Dir.chdir(option_working_directory)
         gen_cmd = "puppet module generate #{name}"
         Open3.popen3(gen_cmd) do |stdin, stdout, _stderr|
-          stdout.sync = true
-          ConfigFileImportCommand.interview_questions.each do |q|
-            rd = ''
-            until rd.include? '?'
-              rd = stdout.readline
-              #debug "Read #{rd}"
+          begin
+            stdout.sync = true
+            ConfigFileImportCommand.interview_questions.each do |q|
+              rd = ''
+              until rd.include? '?'
+                rd = stdout.readline
+                debug "Read #{rd}"
+              end
+              answer = @interview_answers[q].gsub('#{module_name}', module_name)
+              stdin.puts(answer)
             end
-            answer = @interview_answers[q].gsub('#{module_name}', module_name)
-            stdin.puts(answer)
+          rescue EOFError => e
+            debug 'Done reading'
+            break
           end
-          rd = ''
           begin
+            rd = ''
             rd = stdout.readline while rd
           rescue EOFError
             debug 'Done reading'
           end
         end
+        Dir.chdir(pwd)
 
         # Now that we have generated the module, add a 'description' to the
         # metadata.json file found at option_working_dir/<name>/metadata.json
         metadata_path = File.join(File.join(option_working_directory, name), 'metadata.json')
-        answer = @interview_answers['description'].gsub('#{module_name}', module_name)
-        sed_cmd = "sed -i '\/\"summary\":\/a \\ \\ \"description\": \"#{answer}\",' #{metadata_path}"
-        debug "About to issue #{sed_cmd}"
-        system sed_cmd
-        report_summary :wrote, :puppet_modules
+        if File.exist? metadata_path
+          answer = @interview_answers['description'].gsub('#{module_name}', module_name)
+          sed_cmd = "sed -i '\/\"summary\":\/a \\ \\ \"description\": \"#{answer}\",' #{metadata_path}"
```

Comment:
Readability does not describe this code. :) At this point I'm not in favor of any chance of adding accidental errors.